### PR TITLE
fix: Solidify and unify Scene disposing logic

### PR DESCRIPTION
**Pull request by mikhail-dcl**
**The main issue:**

- Mutex is not disposed upon scene disposal: it has lead to the threads leak. Should fix the problem with repetitive disposal/scene not loading

**Related smells fixed:**
- Repetitive and/or redundant disposal of APIs, no validation for the aforementioned
- Different logic for intermediate and final scene disposal

### CODE REVIEW

File: Explorer/Assets/Scripts/CrdtEcsBridge/WorldSynchronizer/CommandBufferSynchronizer/SDKComponentCommandBufferSynchronizer.cs [hunk 1]
```diff
@@ -36,6 +36,11 @@ public override void Apply(World world, PersistentCommandBuffer commandBuffer, E
                 case CRDTReconciliationEffect.ComponentModified:
                     // if component is modified then return to the pool the existing one
                     // No need to add it to the command buffer as we already get the component by ref and can "override" it directly without overhead
+                    if (!world.Has<T>(entity))
+                    {
+                        {}
+                    }
```
Comment (Kinerius) [inline Explorer/Assets/Scripts/CrdtEcsBridge/WorldSynchronizer/CommandBufferSynchronizer/SDKComponentCommandBufferSynchronizer.cs:42]: you forgot this?

**Review by Kinerius — APPROVED** [review]
LGTM

**Review by m3taphysics — APPROVED** [review]

**Comment by m3taphysics** [comment]
Disposal appears to be working flawlessly now with the scene always reinitiatlizing and resetting correctly. 

https://drive.google.com/file/d/1DiDGw9TBU8IcR84x6D1-iVLpatY8Bd1g/view?usp=drive_link

**Review by davidejensen — APPROVED** [review]
Tested via unity on mac and it looks fine for me as well!

**Pull request merged**